doc/api/events.md

```diff
@@ -861,6 +861,37 @@ Value: `Symbol.for('nodejs.rejection')`
 
 See how to write a custom [rejection handler][rejection].
 
+## events.on(emitter, eventName)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `emitter` {EventEmitter}
+* `eventName` {string|symbol} The name of the event being listened for
+* Returns: {AsyncIterator} that iterates `eventName` events emitted by the `emitter`
+
+```js
+const { on, EventEmitter } = require('events');
+
+(async () => {
+  const ee = new EventEmitter();
+
+  // Emit later on
+  process.nextTick(() => {
+    ee.emit('foo', 'bar');
+    ee.emit('foo', 42);
+  });
+
+  for await (const event of on(ee, 'foo')) {
+    console.log(event); // prints ['bar'] [42]
+  }
+})();
+```
+
+Returns an `AsyncIterator` that iterates `eventName` events. It will throw
+if the `EventEmitter` emits `'error'`. It removes all listeners when
+exiting the loop.
```

Comment:
The documentation should likely make it clear that only *waiting* for the event is async within a `for await` and that the execution of the inner block is synchronous and will block. It *should* be obvious, but given how many times we've seen people completely abuse and misuse Promises I'd prefer to be explicit.. Otherwise we'll see things like...

```js
const server = ...
for await (const event of on(server, 'request')) {
  // Long blocking bunch of synchronous code
}
```

Reply:
I've added some text in the example. Let me know if it's ok.